app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt

```diff
@@ -40,28 +40,32 @@ class MealViewModel @Inject constructor(private val mealRepo: MealRepository) :
       mealID: Long = _meal.value.mealID,
       mealTemp: Double = _meal.value.mealTemp,
       ingredients: MutableList<Ingredient> = _meal.value.ingredients,
-      nutritionalInformation: NutritionalInformation = _meal.value.nutritionalInformation,
       firebaseID: String = _meal.value.firebaseId,
-      createdAt: LocalDate = _meal.value.createdAt
+      createdAt: LocalDate = _meal.value.createdAt,
+      tags: MutableList<MealTag> = _meal.value.tags
   ) {
+    // When we make a new Meal, we add all the ingredient values into nutritionalInfo. If we pass
+    // existing values, then its double adding
```

Comment:
Good catch! This is due to the fact that the init function is called during the creation of a new Meal, which causes the nutritional values to be automatically updated according to the ingredients.